sources/src/parser/geocoordinatetransformer.cpp

```diff
@@ -100,6 +104,14 @@ class GeoTransform
         OGRErr err = sourceSRS.SetFromUserInput(sourceURN.c_str());
         sourceSRS.SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
 
+        // Guess the units of the vertical axis for 2D CRS
+        double verticalToMeters = 1.0;
+        int axesCount = sourceSRS.GetAxesCount();
+        if (axesCount == 2)
+        {
+            verticalToMeters = sourceSRS.GetLinearUnits();
```

Comment:
I don't know GDAL very well, what if the SRS is a cartographic SRS so the first two axes are angles? what will it return then?

Reply:
There's separate getters for angular units and linear units. I believe linear units will default to meters if that's the case. I should confirm that this happens with point clouds.

Reply:
https://gdal.org/doxygen/classOGRSpatialReference.html#aefbcac403652e15cfcb1d864d84a3706 -- looks like it does default to 1.0 (meters).

Also it seems like this is deprecated in favor of a version that returns the units name